src/QueryDescription/index.js

```diff
@@ -231,7 +264,32 @@ export const on: OnFunction = (table, leftOrWhereDescription, valueOrComparison)
 }
 
 const syncStatusColumn = columnName('_status')
-const getJoins: (Condition[]) => [On[], Where[]] = (partition(propEq('type', 'on')): any)
+const extractClauses: (Condition[]) => QueryDescription = conditions => {
+  const clauses = { join: [], sortBy: [], where: [], take: null, skip: null }
+  conditions.forEach(cond => {
+    let { type } = cond
+    switch (type) {
+      case 'on':
+        type = 'join'
+        // fallthrough
+      case 'sortBy':
+        // $FlowFixMe: Flow is too dumb to realize that it is valid
+        clauses[type].push(cond)
+        break
+      case 'take':
+      case 'skip':
+        // $FlowFixMe: Flow is too dumb to realize that it is valid
+        clauses[type] = cond
+        break
+      default:
+      case 'where':
+        clauses.where.push(cond)
+        break
+    }
+  })
```

Comment:
I don't get it… aren't you just pushing to `clauses[newType]` in all cases, just that newType has different names? may be easier to have a helper function with a switch  that returns one of `join`, `sortBy`, etc?

Reply:
The problem is the code is difficult to be statically analyzed for type checking (although type checking for dynamically accessed property as switch condition seems to be broken in Flow, it does work in TypeScript), but maybe the `where` case can be merged into `sortBy` case.

Reply:
refactored to remove repeated push calls